api/managers/user_manager.js

```diff
@@ -1,14 +1,244 @@
-const axios = require('axios');
+// Manager for spotify user related API calls and user data modifications
 
-// spotify user related API calls
+// Datastore Setup
+const project_id = process.env.GCP_PROJECT_ID;
+const Datastore = require('@google-cloud/datastore');
+const datastore = new Datastore({
+  projectId: project_id,
+});
+
+const SPOTIFY_UTILS = require('../utils/spotify_utils');
+const AXIOS = require('axios');
+
+// Max Size for all user fav_artists and fav_genres structures
+const MAX_SIZE = 50;
+
+/*/
+/* Params:
+/*   data_struct: Either fav_artists or fav_genres structures that have the appropriate structure for implementing LRU eviction policy
+/*
+/* Returns:
+/*   The modified data_struct where the global_counter and last_accessed of each item are subtracted by the lowest
+/*   last_accessed value, and weights of all items are divided by the lowest weight that is > 1 (n/1 = n and n/0 = undefined)
+/*/
+normalizeIncrementingValues = (data_struct) => {
+  let lowest_lru_time = data_struct.global_counter // Must be at most global_counter
+  let lowest_weight = Number.MAX_SAFE_INTEGER; // Should be some big int
+
+  for (var key in data_struct.items) {
+    if (data_struct.items[key].last_accessed < lowest_lru_time) lowest_lru_time = data_struct.items[key].last_accessed;
+
+    if (data_struct.items[key].weight < lowest_weight) lowest_weight = data_struct.items[key].weight;
+  }
+
+  data_struct.global_counter -= lowest_lru_time;
+  for (var key in data_struct.items) {
+    data_struct.items[key].last_accessed -= lowest_lru_time;
+  }
+
+  if (lowest_weight > 1) {
+    for (var key in data_struct.items) {
+      data_struct.items[key].weight /= lowest_weight;
+    }
+  }
+
+  return data_struct;
+}
+
+/*/
+/* Params:
+/*   data_struct: Either fav_artists or fav_genres structures that have the appropriate structure for implementing LRU eviction policy
+/*   key: The key that should be inserted into the structure
+/*
+/* Returns:
+/*   The modified data_struct where the key is inserted into the structure. If the structure was initially full,
+/*   the inserted key will replace a key in the structure that was accessed the least recently
+/*/
+insertWithLruPolicy = (data_struct, key) => {
```

Comment:
use timestamps instead of global_counter and last_accessed and simplify the structures in datastore